WalletWasabi.Fluent/Models/UI/UiContext.cs

```diff
@@ -29,11 +39,12 @@ public UiContext(IQrCodeGenerator qrCodeGenerator, IQrCodeReader qrCodeReader, I
 	public IFileSystem FileSystem { get; }
 	public IClientConfig Config { get; }
 	public IApplicationSettings ApplicationSettings { get; }
+	public IExchangeRateProvider ExchangeRateProvider { get; }
 
-	/// <summary>
-	/// The use of this property is a temporary workaround until we finalize the refactoring of all ViewModels (to be testable)
-	/// </summary>
-	public static UiContext Default;
+	public BtcAmount CreateAmount(Money? money)
+	{
+		return new BtcAmount(money ?? Money.Zero, ExchangeRateProvider);
+	}
```

Comment:
I'd rather not have this method in the `UiContext` class itself, but rather inside a service.

Something like:

`var amount = UiContext.AmountProvider.Create(money);`

Also, don't forget to rename `IExchangeRateProvider` to `IAmountProvider`, because the former already exists in the backend and we don't want to introduce a name clash. Not a compilation problem because they're in different namespaces, but still confusing for people who look for these interfaces using <key>Ctrl</key> <key>.</key>

Reply:
I didn't include a service because it's currently only a method. 
Regarding the rename of `IExchangeRateProvider` to `IAmountProvider`. I think we still need an IExchangeRateProvider for scenarios like this:

https://github.com/zkSNACKs/WalletWasabi/blob/c2774d2af624a02796ffd6c8082ad31adbe051f5/WalletWasabi.Fluent/ViewModels/Wallets/Home/Tiles/BtcPriceTileViewModel.cs#L5C2-L5C2

We don't need an amount here, but only the exchange rate.

Reply:
@SuperJMN Also, in most cases we do have an `IWalletModel`, except for specific places like `TransactionBroadcasterViewModel`.

It's probably a good idea to add `IWalletModel.CreateAmount(Money money)` method as well.

Reply:
> @SuperJMN Also, in most cases we do have an `IWalletModel`, except for specific places like `TransactionBroadcasterViewModel`.
> 
> It's probably a good idea to add `IWalletModel.CreateAmount(Money money)` method as well.

OK!

BTW, please, check `BtcPriceTileViewModel.cs`, where only the exchange rate is used, not a full amount.

Reply:
> @SuperJMN How about moving the exchange rate observable to `IAmountProvider`? That way you only have 1 interface and you still keep the ability to provider exchange rate to ViewModels that need it.

That makes me happier, but the name might be confusing. I wonder if there's a good name for the abstraction.

- It deals with amounts in BTC and USD.
- It provides exchange rates

Reply:
@SuperJMN 

> I wonder if there's a good name for the abstraction

There are 2 hard things in computer science:

 - naming things `<- you are here`
 - cache misses
 - off by one errors

Reply:
I've renamed **BtcAmount** to just **Amount**. I think it plays well. What are your feelings @zkSNACKs/ui-team ?